Automattic/wp-calypso
# Fix: Show wpcom eCommerce plan on dev/stage environments, and hide Jetpack Personal on production

### Changes proposed in this Pull Request

* Fixes the issue introduced by https://github.com/Automattic/wp-calypso/pull/44313 that hides the wpcom eCommerce plan in dev and staging environments. 
* Introduces a new config feature `jetpack/personal-plan` that will be used to hide the Personal plan. Right now, Jetpack personal will be hidden on dev and staging only.

#### WPCOM BEFORE - eCommerce plan missing in dev and staging** 

**Plans page**

<img width="1110" alt="Screenshot 2020-07-29 at 11 16 30 AM" src="https://user-images.githubusercontent.com/1269602/88771925-9817f900-d19d-11ea-8ed2-71b7c8949e76.png">

**Signup flow**

<img width="1669" alt="Screenshot 2020-07-29 at 11 14 37 AM" src="https://user-images.githubusercontent.com/1269602/88771958-a36b2480-d19d-11ea-856d-48c647ed018f.png">


#### WPCOM AFTER** 

<img width="1672" alt="Screenshot 2020-07-29 at 1 10 34 PM" src="https://user-images.githubusercontent.com/1269602/88772013-baaa1200-d19d-11ea-917f-15f6cc4550e3.png">
<img width="1102" alt="Screenshot 2020-07-29 at 1 10 16 PM" src="https://user-images.githubusercontent.com/1269602/88772022-bed62f80-d19d-11ea-84fd-d6fe51849200.png">


#### JETPACK BEFORE - Personal plan visible in production wordpress.com (not proxied) ** 

<img width="1470" alt="Screenshot 2020-07-29 at 1 13 22 PM" src="https://user-images.githubusercontent.com/1269602/88772117-e3caa280-d19d-11ea-9d29-3f1f1130d9f8.png">

#### JETPACK AFTER - Personal plan hidden ** 

<img width="892" alt="Screenshot 2020-07-29 at 1 11 14 PM" src="https://user-images.githubusercontent.com/1269602/88772138-efb66480-d19d-11ea-8d55-81fc0544ecfe.png">



#### Testing instructions

1. Verify that the testing instructions in https://github.com/Automattic/wp-calypso/pull/44313 pass.
2. We need to verify that Jetpack Personal is hidden in production. To test this, we will have to change environment and I think that needs some docker setup? IMO if testing is too complicated, we should just deploy and check directly on production.
3. On wpcom proxied, verify that the ecommerce plan shows up in the signup flow and in Calypso's /plans page.

Fixes #
